src/frontend/src/binder/expr/mod.rs

```diff
@@ -478,7 +485,63 @@ impl Binder {
             .map(|expr| self.bind_expr_inner(*expr))
             .transpose()?;
 
+        let mut constant_lookup_inputs = Vec::new();
+
+        // The prerequisite for optimization
+        let mut optimize_flag = conditions.len() >= CASE_WHEN_ARMS_OPTIMIZE_LIMIT;
+
+        // Insert a dummy expression to avoid being constant fold
+        // if eventually turn into `ConstantLookupExpression` by optimizing
+        match operand {
+            Some(ref t) => {
+                let Ok(bind_result) = self.bind_expr_inner(Expr::BinaryOp {
+                    left: t.clone(),
+                    op: BinaryOperator::Eq,
+                    right: t.clone(),
+                }) else {
```

Comment:
Is `operand == operand` the dummy expression?  Why do we need this?

Reply:
Yes. The reason we need this is that, without the dummy expression inside `constant_lookup_inputs`, during later optimization phase all the arms will be constant folding, which will then lead to, during runtime execution (will directly go through `eval_row`), we'll not have the ability to check the input and perform constant lookup, and this is not acceptable.

Reply:
I still don't quite get that. IIUC, constant folding needs to run `eval_row` with an empty chunk. In that case, the first elements of `constant_lookup_inputs` should be a constant, we can evaluate it, right? The dummy expression seems unnecessary.

Reply:
> The dummy expression seems unnecessary.

Without the dummy expression, the check at `try_fold_const` will success, and the data flow will directly go through `eval_row` in backend, in which case we can't get the input `DataChunk` and do the constant lookup.

Reply:
In that case, `try_fold_const` has ensured all expressions are constant, so there is no need to evaluate any real data chunk. Maybe we can assert the input row is empty to avoid misuse? Or implement the `eval
_row` as `eval`.